provisioning/miniprov/test/test_cdf.py

```diff
@@ -107,6 +124,12 @@ def ret_values(value: str) -> Any:
                 1,
                 'cluster>srvnode_1>network>data>interface_type':
                 'o2ib',
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>site>storage_set_count': 1,
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set>server_node_count': 1,
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>name': 'StorageSet-1',
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>server_nodes': ['srvnode_1'],
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>durability>data': 1,
+                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>durability>parity': 0
```

Comment:
As far as I understand, there is no test case that checks that DiskRef, PoolDesc or ProfileDesc is actually generated by the given ConfStore values.

Here is the example from the existing tests:
```python
    def test_provided_values_respected(self):
        store = ValueProvider()

        def ret_values(value: str) -> Any:
            data = {
                'cluster>server_nodes': {
                    "blah": "srvnode_1"
                },
                'cluster>srvnode_1>hostname':
                'myhost',
                'cluster>srvnode_1>storage>data_devices': ['/dev/sdb'],
                'cluster>srvnode_1>network>data>private_interfaces':
                ['eth1', 'eno2'],
                'cluster>srvnode_1>storage>metadata_devices': ['/dev/meta'],
                'cluster>srvnode_1>s3_instances':
                1,
                'cluster>srvnode_1>network>data>interface_type':
                'o2ib',
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>site>storage_set_count': 1,
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set>server_node_count': 1,
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>name': 'StorageSet-1',
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>server_nodes': ['srvnode_1'],
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>durability>data': 1,
                'cluster>92f444df-87cc-4137-b680-aab3b35d1695>storage_set1>durability>parity': 0
            }
            return data[value]

        store._raw_get = Mock(side_effect=ret_values)

        ret = CdfGenerator(provider=store)._create_node_descriptions()
        self.assertIsInstance(ret, list)
        self.assertEqual(1, len(ret))
        self.assertEqual(Text('myhost'), ret[0].hostname)
        self.assertEqual(Text('eth1'), ret[0].data_iface)
        self.assertEqual(1, ret[0].s3_instances)
```
(note the lines like this: `self.assertEqual(Text('eth1'), ret[0].data_iface)` - Text is our own type that will be translated into Dhall expression; you added more such types but there is no test that checks that instances of such types ever get created)